File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/DependencyInjectionAfter/_1_ControlFreakRefactoredToDependencyInjection/Interfaces/AcmeMessage.cs
using DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Services;

namespace DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Interfaces
{
  public interface AcmeMessage
  {
    void AuthorizeUsing(IAuthorization authorizationRules);
    void WriteTo(DataDestination dataDestination);
  }
}

File: Java/dependency-injection/_03_control-freak-refactored/src/main/java/com/github/grzesiek_galezowski/DependencyInjectionAfter/Interfaces/AcmeMessage.java
package com.github.grzesiek_galezowski.DependencyInjectionAfter.Interfaces;

import com.github.grzesiek_galezowski.DependencyInjectionAfter.Services.Authorization;

public interface AcmeMessage {
  void authorizeUsing(Authorization authorizationRules);

  void writeTo(DataDestination dataDestination);
}
